src/Documentation/Event-Sourcing/LogConsistencyProviders.md

```diff
@@ -0,0 +1,54 @@
+---
+layout: page
+title: Included Log-Consistency Providers
+---
+
+# Built-In Log-Consistency Providers
+
+The `Microsoft.Orleans.EventSourcing` package includes several log-consistency providers that cover basic scenarios suitable to get started, and allow some extensibility.
+
+
+### Orleans.EventSourcing.**StateStorage**.LogConsistencyProvider
+
+This provider stores *grain state snapshots*, using a standard storage provider that can be configured independently. 
+
+The data that is kept in storage is an object that contains both the grain state (as specified by the first type parameter to `JournaledGrain`) and some meta-data (the version number, amd a special tag that is used to avoid duplication of events when storage accesses fail).
+
+Since the entire grain state is read/written every time we access storage, this provider is not suitable for objects whose grain state is very large. 
+
+This provider does not support `RetrieveConfirmedEvents`: it cannot retrieve the events from storage because the events are not persisted.
```

Comment:
> This seems like an advanced scenario. There is currently no way to write such a provider in a general way (since it requires knowing all the grains and how the grain state was changed).

I agree... Use cases for that, like @jdom mentioned, is if I add a new system that want to leverage all history of an already in place system. Today, what I do (since I'm using SS and not the storage provider from Orleans), what I do is make a console App that read whole stream of events up to a certain part directly from ATS with StreamStone, and publish then to a stream which the new system will consume and generate their projections.

Re. Checkpoints... I think at user level you should have a `WriteCheckpointAsync()` so the user decide when to create a checkpoint for a given stream. 

The major problem with the current storage APIs is that it is unreadable outside Orleans. And that is why you need to expose APIs like the `GetConfirmedEvents()` as I requested so we can support the scenario I mentioned here first. If we have a *open* storage, like SS does, or use GES, people can easily read whole stream and publishe somewhere else.  I'll post my API suggestion for ES storage in #2723 later on.